src/io/mod.rs

```diff
@@ -186,6 +194,25 @@ impl<D: DeserializeOwned> DelimFileReader<D> {
     pub fn read(&mut self) -> Option<Result<D>> {
         self.record_iter.next().map(|result| result.map_err(FgError::ConversionError))
     }
+
+    fn validate_header(header: &StringRecord, delimiter: u8) -> Result<()> {
+        let delim = String::from_utf8(vec![delimiter]).unwrap();
+        let found_header_parts: Vec<&str> = header.iter().collect();
+        let expected_header_parts = serde_aux::prelude::serde_introspect::<D>();
+
+        // Expected header fields must be a _subset_ of found header fields
+        let ok = expected_header_parts.iter().all(|field| found_header_parts.contains(field));
+
+        if !ok {
+            let expected = expected_header_parts.join(&delim);
+            return Err(FgError::DelimFileHeaderError {
+                expected,
+                found: header.as_slice().to_owned(),
```

Comment:
This is buggy... If the header contains the expected delimiter the strings are just concatenated sans delimiter. E.g.

`foo, bar` => `foobar`.

Reply:
Fixed this and added new tests.